src/index.ts

```diff
@@ -251,12 +252,18 @@ export function withSpid({
         middlewareCatchAsInternalError((req, res, next) => {
           fromNullable(req.query)
             .mapNullable(q => q.authLevel)
+            // As only strings can be key of SPID_LEVELS record,
+            //  we have to narrow type to have the compiler accept it
+            // In the unlikely case authLevel is not a string, an empty value is returned
+            .filter((e): e is string => typeof e === "string")
             .chain(authLevel =>
               lookup(authLevel, SPID_LEVELS).map(_ => authLevel)
             )
             .chain(
-              fromPredicate(authLevel =>
-                appConfig.spidLevelsWhitelist.includes(authLevel)
+              fromPredicate(
+                authLevel =>
+                  t.keyof(SPID_LEVELS).is(authLevel) &&
```

Comment:
If we check here the value of `authLevel` with `t.keyof(SPID_LEVELS).is(authLevel)` maybe we can remove the chain/lookup before or remove the `is` statement adding a casting like
```typescript
.chain(authLevel =>
  lookup(authLevel, SPID_LEVELS).map(_ => authLevel as keyof SPID_LEVELS)
)
```

Reply:
Nice point, I did this: https://github.com/pagopa/io-spid-commons/pull/88/commits/2a9c5715e069e18e760868e5cda43ea3840e9ac4

Reply:
Nice point, I did this: https://github.com/pagopa/io-spid-commons/pull/88/commits/b416411d3e3db6a64a4d1805d33b8e88548be1ab